### PR TITLE
Refactoring task files

### PR DESCRIPTION
**Pull request by shubham76**
This is regarding #3775.
Tried only moving `fileify` function as mentioned by @agjohnson [here](https://github.com/rtfd/readthedocs.org/issues/3775#issuecomment-373555709) as it operates on versions.
Is something like this expected?
I only tried with one function. If things are right here I'll go ahead and make other changes.

### CODE REVIEW

File: readthedocs/builds/tasks.py [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+                                                  DockerBuildEnvironment)
+from readthedocs.doc_builder.exceptions import BuildEnvironmentError
+from readthedocs.doc_builder.loader import get_builder_class
+from readthedocs.doc_builder.python_environments import Virtualenv, Conda
+from readthedocs.projects.models import APIProject
+from readthedocs.restapi.client import api as api_v2
+from readthedocs.restapi.utils import index_search_request
+from readthedocs.search.parse_json import process_all_json_files
+from readthedocs.vcs_support import utils as vcs_support_utils
+from readthedocs.worker import app
```
Comment (agjohnson) [inline readthedocs/builds/tasks.py]: It looks like almost all of these imports are not used, these should be clean up.

File: readthedocs/builds/tasks.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+
+log = logging.getLogger(__name__)
+
+HTML_ONLY = getattr(settings, 'HTML_ONLY_PROJECTS', ())
```
Comment (agjohnson) [inline readthedocs/builds/tasks.py]: This doesn't look used either.

File: readthedocs/builds/tasks.py (hunk 1 above)
Comment (shubham76) [reply readthedocs/builds/tasks.py]: Yes. I will do that. :+1:

File: readthedocs/builds/tasks.py (hunk 2 above)
Comment (shubham76) [reply readthedocs/builds/tasks.py]: Yes. Right. :+1:

**Comment by shubham76** [comment]
Hey @agjohnson! I completed refactoring other functions and made changes as you had requested. Could you take a look and let me know if something else needs to be done?
Thank you for your time! :)

**Comment by shubham76** [comment]
Hey @humitos! I did the changes as you had requested. Could you take a look? 
Also, could you help me fix those linting errors?

**Comment by humitos** [comment]
@shubham76 change look good! Could you please solve the conflics by rebasing your branch? After that, could you check if the tests need to be updated also? Thanks!

**Comment by ericholscher** [comment]
This has a ton of conflicts in it, so I'm going to close it. It also combined a refactor with some feature additions, where I'd much prefer just a straight refactor, and then an additional PR with additions. Thanks for opening it, but redoing this will probably require a new PR anyway because of all the conflicts.

**Pull request closed without merging**